File: CreationalPatterns/Prototype/Java/Registry/PizzaRegistry.java
package Registry;

import java.util.HashMap;
import java.util.Map;

import Prototype.IPizzaPrototype;

public class PizzaRegistry {
    private Map<String, IPizzaPrototype> pizzaPrototypes = new HashMap<>();

    public IPizzaPrototype createPizza(String key){
        IPizzaPrototype prototype = pizzaPrototypes.get(key);
        if (prototype == null) {
            throw new IllegalArgumentException("Pizza type '" + key + "' not found.");
        }
        return prototype.copy();
    }

    public void addPizza(String key, IPizzaPrototype pizza){
        pizzaPrototypes.put(key, pizza);
    }
}

File: CreationalPatterns/Prototype/C#/Prototype/Registry/PizzaRegistry.cs
public class PizzaRegistry {
    private Dictionary<String, IPizzaPrototype> PizzaPrototypes = new Dictionary<string, IPizzaPrototype>();

    public IPizzaPrototype createPizza(String key){
        IPizzaPrototype? prototype = PizzaPrototypes.GetValueOrDefault(key);
        if (prototype == null) {
            throw new Exception("Pizza type '" + key + "' not found.");
        }
        return prototype.Copy();
    }

    public void addPizza(String key, IPizzaPrototype pizza){
        PizzaPrototypes.Add(key, pizza);
    }
}
